main.py

```diff
@@ -174,25 +200,34 @@ def should_nag(message: discord.Message) -> Optional[NagType]:
     """
     Test whether we should respond to the message.
 
-    Requires a native twitter link. Then, checks at least one of the following conditions is met:
-    1. When Twitter embeds break, always reply
-    2. Both channel and at least one embed are marked sensitive
-    3. Embed is safe and there is a video
+    Requires a native twitter link. Then, checks at least one of the following conditions is met before posting:
+    - If the post has at least one NSFW tweet, channel must be NSFW
+        - When Twitter embeds break, always check Fx for NSFW status
+    - Embed is safe and there is a video
+    - Embed is safe and has multiple images
     """
     if not re.search(r"(//|mobile\.)(twitter|x)\.com", message.content):
         return None
-    if not message.embeds:
-        return NagType.FULL
     # sensitive check _must_ be done before other types
     # to prevent mixing NSFW and SFW links from embedding
-    if any(_is_sensitive_tweet_embed(em) for em in message.embeds):
+    if _has_sensitive_tweet(message):
         channel = message.channel
         if channel.type in (discord.ChannelType.private, discord.ChannelType.group):
             # always allow NSFW in (group) DMs
             return NagType.FULL
-        if hasattr(channel, "nsfw") and getattr(channel, "nsfw") is True:
-            # .nsfw seems to not exist on some channel types
-            # and discord.py type narrowing is limited
+        if (
+            isinstance(
+                channel,
+                (
+                    discord.TextChannel,
+                    discord.ForumChannel,
+                    discord.StageChannel,
+                    discord.VoiceChannel,
+                    discord.Thread,
+                ),
+            )
+            and channel.is_nsfw()
+        ):
```

Comment:
Is this purely for type narrowing to "anything that has the is_nsfw property"? It could possibly use protocols: https://peps.python.org/pep-0544/. Implemented in 3.8, so no issue with Python versions either.
```py
class HasNsfwProperty(typing.Protocol):
    def is_nsfw(self) -> bool: ...
```
Would allow this to become just:
```py
        if isinstance(channel, HasNsfwProperty) and channel.is_nsfw():
```

Reply:
Oh, neat, yeah that was the intention. I think that should be a fine change then.